src/archive.rs

```diff
@@ -53,15 +53,10 @@ impl CompressionMethod {
         }
     }
 
-    pub fn is_enabled(
-        self,
-        tar_enabled: bool,
-        tar_archive_enabled: bool,
-        zip_enabled: bool,
-    ) -> bool {
+    pub fn is_enabled(self, tar_enabled: bool, tar_gz_enabled: bool, zip_enabled: bool) -> bool {
         match self {
+            CompressionMethod::TarGz => tar_gz_enabled,
```

Comment:
Come to think of it, it probably wasn't proper of me to go with `CompressionMethod` here as it should really have been `ArchiveMethod` from the beginning. Oh well, might change this after merging your PR. :)